clientv3/retry.go

```diff
@@ -36,14 +36,22 @@ func (c *Client) newRetryWrapper() retryRpcFunc {
 			eErr := rpctypes.Error(err)
 			// always stop retry on etcd errors
 			if _, ok := eErr.(rpctypes.EtcdError); ok {
-				return err
+				if grpc.Code(err) != codes.Unavailable {
```

Comment:
This isn't safe. Retrying will violate at-most-once semantics for Put/Del/Txn since a `request timed out` error could be returned even if the request makes it through raft. This can happen if there's a partition between the time of submitting a raft proposal and commit.